ruby/CMakeLists.txt

```diff
@@ -105,3 +106,72 @@ endif()
 if( BUILD_PAT )
   # don't need modeleditor libs in pat
 endif()
+
+###############################################################################
+#                        T E S T I N G:   C T E S T S                         #
+###############################################################################
+
+# find all tests
+file(GLOB RUBY_TEST_SRC "test/*.rb")
+
+# TODO: It doesn't not work with this executable that's in build/Ruby-install/bin
+# message("RUBY_EXECUTABLE=${RUBY_EXECUTABLE}")
+
+set(_RUBY_POSSIBLE_EXECUTABLE_NAMES
+  ruby
+  ruby2.5
+  ruby2.5.1
+  ruby25
+  ruby251)
+
+# TODO: this isn't great but I haven't found a better way to locate the system ruby (and avoid the one in build/Ruby-install/)
+find_program(SYSTEM_RUBY_EXECUTABLE NAMES ${_RUBY_POSSIBLE_EXECUTABLE_NAMES}
+  HINTS
+    "/usr"
+    "/usr/local"
+    "/usr/share/rvm/rubies/ruby-2.5.1/bin/"
```

Comment:
Should we check for newer ruby first?

Reply:
we should use FindRuby from cmake. I submitted three PRs to cmake to have it work. it's only in cmke 3.18+ though, so perhaps we need to extract the find module and place it in our repo